CONTRIBUTING.md

```diff
@@ -0,0 +1,102 @@
+
+## Contents
+- [Contributing a change to existing workload(s)](#contributing-a-change-to-existing-workloads)
+- [Test changes](#test-changes)
+    - [Testing changes locally](#testing-changes-locally)
+    - [Testing changes with integration tests](#testing-changes-with-integration-tests)
+- [Publish changes in a pull-request](#publish-changes-in-a-pull-request)
+- [Reviewing pull-requests](#reviewing-pull-requests)
+     - [Backporting](#backporting)
+     - [Important note on backporting reverted commits](#important-note-on-backporting-reverted-commits)
+- [Contributing a workload](#contributing-a-workload)
+
+
+## Contributing a change to existing workload(s)
+
+Before making a change, we recommend you fork the official workloads repository and make the change there.
+
+You should also consider whether or not your change should be applied to one or more branches.
+- If you know your change is only applicable to a specific branch, create the feature branch based off of that specific branch and make the changes there.
+- If the change is applicable to several branches, create the feature branch based off of `main` branch and make the changes there.
+
+## Test changes
+
+Once you've made your change in your feature branch, we recommend testing it locally and with integration tests via your forked OpenSearch Benchmark repository.
+
+### Testing changes locally
+It's recommended to test your change locally by performing the following:
+1. Set up or use an existing OpenSearch cluster to test your changes against.
+2. Based on the major version of the test cluster, cherry-pick the commit(s) with your change to the corresponding major version branch in your forked workloads repository. For example, if you're testing against an OpenSearch 2.X.X cluster, cherry-pick the changes from the feature branch to `2` branch.
+3. Run the OpenSearch Benchmark command and against your cluster in `--test-mode`. Ensure it works successfully.
+
+Other tips when running the command in test mode against your cluster:
+- Ensure you are using the workloads repository that you committed your changes in. To enforce this, provide the path to your repository via the `--workloads-repository` parameter.
+- Alternatively, you can force OSB to use a specific branch by specifying the distribution version of your OpenSearch cluster via the `--distribution-version` parameter. To build on the example from the previous step, to ensure you are using branch `2`, set `--distribution-version=2.0.0` in the OpenSearch Benchmark command.
+
+### Testing changes with integration tests
+
+To ensure that there are no other breaking changes, we recommend testing with your forked OpenSearch Benchmark repository Github Actions.
+
+**Prerequisites:**
+
+Before we can run integration tests, we recommend dedicating a remote branch that you can repeatedly use to run integration tests on your forked workloads repository. You only need to perform these actions once.
```

Comment:
Did you mean the forked opensearch benchmark repository?

Reply:
Ah, you're right. I'll rephrase it to:
```
we recommend dedicating a remote branch in your forked OpenSearch Benchmark repository. You will use this branch whenever you need to run integration tests against changes in your forked workloads repository. 
```